### DISCUSSION TITLE
How to debug further "You don't have permission to access this."?

### DISCUSSION THREAD

**Discussion by Pachat** [directus/directus#15233] [Archive]
One of tens collections has an M2M item.

- Adding an M2M item is OK. 
- But after **modifying** the item I get `You don't have permission to access this.`  on saving the parent item.

I have set`All Access` to all my collections and to all Directus collections.
But still get the error.

Are there other places to track this error?
Is it a bug or should I dig further? Where?

**Comment by rijkvanzanten** [comment]
As of right now, Directus doesn't return any more information on purpose, as that would lead to a security vulnerability in which you could extract all the existing IDs in a collection. There's an open discussion here to potentially add a new option to allow the API to return "friendly" or "dev mode" errors rather than these more secure abstracted away ones here: https://github.com/directus/directus/discussions/4368

**Reply by Pachat** [reply]
To prevent unwanted privacy disclosure, an .env variable could be set only on debugging.

**Reply by pierre-H** [reply]
Hi @rijkvanzanten,
Any news about that? 
This is, in my opinion, the most annoying thing in Directus (which is a great tool !)

**Comment by Pachat** [comment]
A "friendly" notice would greatly help to search what " **this** " is. 
Thank you for pointing me the open discussion.

**Reply by rijkvanzanten** [reply]
Also down to discuss (maybe in a new feature request discussion) to update the "this" verbiage with something a hair more descriptive, while still being secure. Something like `You don't have access to item "5" in "articles"` would already be tremendously more helpful in your case right?

**Reply by Pachat** [reply]
Totally agree.
The main problem I face with many collections and fields is to discover whether the limitation comes from the `collections` (and which one, e.g. the collection, the M2M collection or the junction collection) or if it comes from the `fields ` permissions filters or conditions.

**Comment by asheroto** [comment]
I'm facing this issue as well. Using Windows, just installed. Any ideas?

I even tried connecting to my MariaDB instance using the root user.... same error.

**Reply by azrikahar** [reply]
What was the scenario you were going through leading to this issue? Are you getting this issue in the App itself, or when making requests to the API via postman or other applications? for example, were you making a GET request to get items in a collection? or maybe PATCH/DELETE request for update/delete?

**Reply by asheroto** [reply]
Thanks, I should have been more descriptive in my post.

I had an existing MariaDB database with some data in it on my development environment. I set up the self hosted Directus package using ...

```javascript
npm init directus-project my-project
```

I connected it to my database, but then face this error message when I tried to add an input box  to any of my tables, or when doing anything for the initial configuration

**Reply by azrikahar** [reply]
Are there any more descriptive logs in the terminal? I'm not sure what could be the cause in your setup, but perhaps you can try with another clean MariaDB database just to rule out any other possibilities.

Here's what I tried just now, and it seems to work for me:

1. Spin up a new MariaDB container on another machine in the local network using the following compose file:

    ```yaml
    services:
        dev-mariadb:
            image: mariadb:latest
            container_name: dev-mariadb
            restart: unless-stopped
            ports:
                - 3306:3306
            environment:
                - MYSQL_ROOT_PASSWORD=root
                - MYSQL_USER=directus
                - MYSQL_PASSWORD=password
                - MYSQL_DATABASE=directus_db
    ```

2. Use a db client to connect to the MariaDB using the `directus` user, and run the following statement to create a table:

    ```sql
    create table projects(
        id int auto_increment,
        name varchar(255) not null,
        start_date date,
        end_date date,
        created_at timestamp default current_timestamp,
        primary key(id)
    );
    ```

3. Run `npm init directus-project my-project` locally, connecting to that machine via a local IP, and use the `directus` user, _not_ the root user, as shown below:

    (root user should still work, I'm just stating I'm not using the root user as a demo that it will work as well)
    
    ![WindowsTerminal_stmZqrBDAe](https://user-images.githubusercontent.com/42867097/188068724-913108d8-b50d-49c8-84a8-0c92e49215da.png)

4. cd to `my-project`, run `npx directus start` to start the newly created directus instance

5. Login, and setup the `projects` table I created in step 2, add a `description` field, and create a new record. Everything seems to work fine on my end:

    https://user-images.githubusercontent.com/42867097/188069284-cc1f9a13-fd4d-42e7-8cd5-2627e2c55e67.mp4

**Comment by asheroto** [comment]
Ah, thanks for the tip. I didn't see this before

![image](https://user-images.githubusercontent.com/49938263/188331684-029f3ccd-aa25-43de-b51e-e7aae8a52384.png)

So the issue actually was that there were several primary keys set. I started using [Navicat](https://navicat.com/en/products/navicat-premium) a few weeks ago and was clicking these fields...

https://user-images.githubusercontent.com/49938263/188331870-9d73db44-54b1-4054-a537-52ef848d95e8.mp4

Turns out that actually, for some reason, sets each of them as primary keys instead of indexes.

Once I set only one as the primary key is started working! Yay.

Thank you for replying!

**Comment by bill-o-more** [comment]
so I have this issue while connecting to the existing DB with multiple m2m pivot tables with composite keys. Do I have a way to manage such DB (there are at least 5 of them, and the number will grow)?
